wallet/src/de/schildbach/wallet/ui/WalletUriHandlerActivity.java

```diff
@@ -91,23 +94,12 @@ private void handleIntent(final Intent intent) {
         if (Intent.ACTION_VIEW.equals(action) && Constants.WALLET_URI_SCHEME.equals(scheme)) {
             if (intentUri.getHost().equalsIgnoreCase("brokers")) {
                 if (intentUri.getPath().contains("uphold")) {
-                    String code = intentUri.getQueryParameter("code");
-                    String state = intentUri.getQueryParameter("state");
-                    if (code != null && state != null) {
-                        Intent upholdActivityIntent = new Intent(this, UpholdSplashActivity.class);
-                        upholdActivityIntent.putExtra(UpholdSplashActivity.UPHOLD_EXTRA_CODE, code);
-                        upholdActivityIntent.putExtra(UpholdSplashActivity.UPHOLD_EXTRA_STATE,
-                                intentUri.getQueryParameter("state"));
-                        upholdActivityIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
-                        upholdActivityIntent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
-                        if(isTaskRoot()) {
-                            //I'm in my own task and not the main task
-                            upholdActivityIntent.setAction(UpholdSplashActivity.FINISH_ACTION);
-                            LocalBroadcastManager.getInstance(this).sendBroadcast(upholdActivityIntent);
-                        } else {
-                            startActivity(upholdActivityIntent);
-                        }
-                    }
+                    Intent activityIntent = new Intent(this, WalletActivity.class);
+                    activityIntent.putExtra("uri", intentUri);
+                    activityIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+                    activityIntent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
+                    activityIntent.setAction(UpholdPortalFragment.AUTH_RESULT_ACTION);
+                    LocalBroadcastManager.getInstance(this).sendBroadcast(activityIntent);
```

Comment:
Propagating an intent with `LocalBroadcastManager` proved to be the only way to pass the result back to `UpholdFragment` without any side effects (e.g. duplicating instances of the app). I played around with `launchMode` but to no avail.

Perhaps once we move more screens under a single activity we can re-evaluate this class and move the scheme URL handling into the `WalletActivity`.